riptide-stream/src/main/java/org/zalando/riptide/stream/StreamSpliterator.java

```diff
@@ -86,4 +86,13 @@ public long estimateSize() {
     public int characteristics() {
         return ORDERED | IMMUTABLE;
     }
+
+    @Override
+    public void close() throws UncheckedIOException {
```

Comment:
I'm actually not 100% sure that calling JsonParser#close is enough. Depending on configuration it may not close the stream, so we may need to do it here.